crates/partition-store/src/outbox_table/mod.rs

```diff
@@ -82,14 +82,35 @@ fn get_outbox_message<S: StorageAccess>(
 fn truncate_outbox<S: StorageAccess>(
     storage: &mut S,
     partition_id: PartitionId,
-    seq_to_truncate: Range<u64>,
+    range: RangeInclusive<u64>,
 ) {
-    let mut key = OutboxKey::default().partition_id(partition_id);
-    let k = &mut key;
+    let _x = RocksDbPerfGuard::new("truncate-outbox");
+
+    let mut keys = vec![];
+    {
+        let start_key = OutboxKey::default()
+            .partition_id(partition_id)
+            .message_index(*range.start());
+
+        let end_key = OutboxKey::default()
+            .partition_id(partition_id)
+            .message_index(*range.end());
+
+        let mut iter = storage.iterator_from(TableScan::KeyRangeInclusiveInSinglePartition(
+            partition_id,
+            start_key,
+            end_key,
+        ));
+
+        iter.seek_to_last();
+        while iter.valid() {
+            keys.push(iter.key().unwrap().to_vec());
+            iter.prev();
+        }
+    }
 
-    for seq in seq_to_truncate {
-        k.message_index = Some(seq);
-        storage.delete_key(k);
+    for key in keys {
+        storage.delete_cf(Outbox, key);
```

Comment:
The outbox could actually be a candidate for https://github.com/facebook/rocksdb/wiki/Single-Delete which would avoid the propagation of the tombstones to the lowest level.

Reply:
This is unrelated to this PR, though.

Reply:
TIL! Going to read up on this, thank you for the link!

Reply:
I didn't think you're suggesting that we this in this PR - let me know if I misunderstood. I didn't see that we use this anywhere else and I don't fully grok the implications of switching TBH.